packages/components/src/internal/components/forms/model.ts

```diff
@@ -300,24 +300,37 @@ function initGroupByColumn(queryInfo: QueryInfo, column?: string): string {
     return groupByColumn;
 }
 
-export async function initSelect(props: QuerySelectOwnProps): Promise<QuerySelectModel> {
+function queryColumnNames(
+    queryInfo: QueryInfo,
+    displayColumn: string,
+    valueColumn: string,
+    requiredColumns: string[],
+    groupByColumn: string
+): string[] {
+    const queryColumns = queryInfo.pkCols.concat([displayColumn, valueColumn].concat(requiredColumns));
+    const lookupViewColumns = queryInfo.getLookupViewColumns();
+
+    if (groupByColumn) {
+        queryColumns.push(groupByColumn);
+    }
+
+    if (lookupViewColumns.length > 0) {
+        return lookupViewColumns.map(c => c.fieldKey).concat(queryColumns);
```

Comment:
do we need to remove duplicates in this return case, like below?

Reply:
Good call. I've revised the logic to a single return statement.